src/addon/ESMX/Driver/CMakeLists.txt

```diff
@@ -65,6 +65,11 @@ else()
   set(ESMX_EXE_NAME ${ESMX_EXE_NAME} PARENT_SCOPE)
 endif()
 
+# define ESMX_SDIR
+if(NOT DEFINED ESMX_SDIR)
+  set(ESMX_SDIR ${CMAKE_SOURCE_DIR})
```

Comment:
I'm trying to figure out if this makes sense or if it needs to be `set(ESMX_SDIR ${CMAKE_CURRENT_SOURCE_DIR})` or `set(ESMX_SDIR ${CMAKE_CURRENT_LIST_DIR})` or `get_filename_component(ESMX_SDIR ${ESMX_BUILD_FILE} DIRECTORY)`. The last one uses the directory where the esmxBuild file is located.

Reply:
`${CMAKE_CURRENT_SOURCE_DIR})` might also satisfy your need while allowing the ESMX build to be embedded deeper into a cmake build, say for a build that has multiple applications. Do you think the ESMX build will always look for the source from the root CMAKE source directory or from the directory calling the ESMX build. ESMX_Builder is more similar to the latter, which assume the sources are in the directory in which the ESMX build is called.

Reply:
The `CMAKE_CURRENT_SOURCE_DIR` will point to the directory that contains the ESMX Driver `CMakeLists.txt` because it is hooked into the application CMake lists file with `add_subdirectory()`. The case you are describing of using ESMX to build an application within an even larger CMake based build system is still achievable by setting the `ESMX_SDIR` variable explicitly to `CMAKE_CURRENT_LIST_DIR` from the level that contains the `add_subdirectory()` for ESMX.
Bottom line, I think what's in the PR now is the correct way. But it's nice to have thought through all of these scenarios!